server/utils.py

```diff
@@ -164,87 +161,62 @@ def remove_from_uploads(tempdir):
-                'readset': file_metadata[4],
-                'platform': file_metadata[5],
-                'run-type': file_metadata[6],
-                'capture-kit': file_metadata[7],
-                'library': file_metadata[8],
-                'reference': file_metadata[9]
+def get_user_files(user_id, status):
+    files = File.query.filter_by(user_id=user_id, upload_status=status).all()
+    db_files = {}
+    for row in files:
+        db_files[row.identifier] = {
+                'identifier': row.identifier,
+                'sample-name': row.sample_name,
+                'filename': row.filename,
+                'type': row.file_type,
+                'readset': row.readset,
+                'platform': row.platform,
+                'run-type': row.run_type,
+                'capture-kit': row.capture_kit,
+                'library': row.library,
+                'reference': row.reference
             }
-        return files
-    return dict(db_files)
+    return db_files
 
 
 def get_user_by_auth_token(auth_token):
-    if g.db.execute('select exists(select 1 from access where auth_token=? LIMIT 1)', (auth_token,)).fetchone()[0]:
-        return g.db.execute('select user_id from access where auth_token=?', (auth_token,)).fetchone()[0]
-
-
-def insert_file_metadata(form_dict):
-    # Values that need to be inserted/updated in the db
-    update_keys = ['auth_token', 'identifier', 'sample_name', 'filename', 'total_size', 'file_type', 'readset',
-                   'platform', 'run_type', 'capture_kit', 'library', 'reference']
-    update_dict = {
-        'server_id': g.db.execute('select server_id from access where auth_token=?',
-                                  (form_dict['auth_token'],)).fetchone()[0],
-        'user_id': get_user_by_auth_token(form_dict['auth_token']),
-        'date_upload_start': dt.datetime.today().strftime("%Y-%m-%dT%H:%M:%SZ"),
-        'upload_status': 'ongoing'
-    }
-
-    for key in update_keys:
-        update_dict[key] = form_dict[key]
-
-    keys = sorted(update_dict)
-    values = [update_dict[key] for key in keys]
-
-    # If the identifier exists, update the metadata
-    SQL = 'INSERT OR REPLACE INTO files ({}) VALUES ({})'.format(', '.join(keys), ','.join('?' * len(keys)))
-    g.db.execute(SQL, values)
-    g.db.commit()
-
-
-def update_file_metadata(form_dict):
-    # Values that need to be updated in the db
-    update_keys = ['sample_name', 'filename', 'total_size', 'file_type', 'readset', 'platform', 'run_type',
-                   'capture_kit', 'library', 'reference']
-    update_dict = {}
-    for key in update_keys:
-        update_dict[key] = form_dict[key]
-
-    g.db.execute('update files set {} where identifier=?'
-                 .format(', '.join('{0}="{1}"'.format(item[0], item[1]) for item in update_dict.items())),
-                 (form_dict['identifier'],))
-    g.db.commit()
+    return Access.query.filter_by(auth_token=auth_token).first().user_id
```

Comment:
This will crash if the auth token is invalid. Should return None instead. Also, should probably just return the User object, as will happen if it's a foreignkey.